src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -836,14 +838,15 @@ impl_install_tree (RpmOstreeTreeComposeContext *self,
    * if one uses `check-passwd: { "type": "previous" }`. There are a few other optimizations
    * too, e.g. using the previous SELinux policy in unified core. Also, we might need the
    * commit *object* for next version incrementing. */
-  if (self->ref)
+  const char *previous_ref = self->ref ?: opt_previous_commit;
+  if (previous_ref)
     {
-      if (!ostree_repo_resolve_rev (self->repo, self->ref, TRUE,
+      if (!ostree_repo_resolve_rev (self->repo, previous_ref, TRUE,
```

Comment:
In the case of `--previous-commit`, shouldn't we set `allow_noent` to `FALSE`?

Reply:
Yeah fair, it is clearer as separate conditionals.  Done.